src/include/daos/placement.h

```diff
@@ -63,7 +63,9 @@ struct pl_obj_shard {
 	uint32_t	po_shard;	/* shard identifier */
 	uint32_t	po_target;	/* target id */
 	uint32_t	po_fseq;	/* The latest failure sequence */
-	uint32_t	po_rebuilding:1, /* rebuilding status */
+	uint16_t	po_rank;	/* The rank on which the shard exists */
```

Comment:
I am not sure if it is a good idea to define it as 16 bits, the number of engines possibly > 64K in near future?
or just enlarge the structure with more 8B

Reply:
That's good question. Honestly, I am not sure for this. Enlarging it is possible, but almost waste. We can consider to use 20bits (to support 1M ranks) for that? @gnailzenh what's your idea?